src/views/MainView.js

```diff
@@ -100,8 +101,31 @@ const MainView = (props) => {
     )
   };
 
+  // initialize Spotify web player
+  const initializePlayer = () => {
+    window.onSpotifyWebPlaybackSDKReady = () => {
+      const player = new window.Spotify.Player({
+        name: 'encore.',
+        getOAuthToken: callback => {
+          // fetch a new token every time the old one expires
+          props.dispatch(fetchAuthToken(user))
+            .then(res => {
+              let token = res.authToken;
+              callback(token)
+            });
+        }
+      });
+      player.connect();
+      // notify backend about newly connected device as soon as it's ready
+      player.addListener('ready', () => props.dispatch(setSyncMode(user, user.syncMode)));
+    };
+  };
+
+  // defining the Spotify script here as opposed to index.html ensures that window.onSpotifyWebPlaybackSDKReady is
+  // defined when the script is loaded.
   return (
     <div className="MainView">
+      <Script url="https://sdk.scdn.co/spotify-player.js" onLoad={initializePlayer}/>
```

Comment:
maybe move all SDK logic into a `SpotifyPlayer` component that is rendered in `<App>`

Reply:
if you only load the player in `MainView` the music will always stop if you go to `/learn-more`